engine/common/rpc/errors.go

```diff
@@ -10,6 +11,41 @@ import (
 	"github.com/onflow/flow-go/storage"
 )
 
+// ConvertError converts a generic error into a grpc status error
+func ConvertError(err error, msg string, defaultCode codes.Code) error {
```

Comment:
This is a really good point about storage errors, and I'll make that change. Not all storage errors are `NotFound`, and codifying that will create bugs in the future.

The intention with this method is to simplify converting common error conditions into their corresponding grpc status codes immediately before returning them back to the client. Ultimately, this helps with monitoring and alerting nodes since `Internal` and `Unknown` errors should only occur in genuine failure cases.